docs/source/guide/Modules.lrad

```diff
@@ -0,0 +1,70 @@
+# Modules
+
+Modules are dicts with some metadata about a module, and an `:env` key
+containing a radicle environment.
```

Comment:
Can we move this description of the implementation somewhere to the bottom. We could lead with a more generic description of what modules are.